File: content/snippets/go/language_basics/command_line_args.go
package main

import (
	"fmt"
	"os"
)

func main() {
	name := "world"
	if len(os.Args) > 1 {
		name = os.Args[1]
	}

	excited := len(os.Args) > 2 && os.Args[2] == "--excited"
	message := fmt.Sprintf("Hello, %s", name)
	if excited {
		message += "!"
	}

	fmt.Println(message)
}

File: content/snippets/typescript/language_basics/command_line_args.ts
#!/usr/bin/env ts-node

import { setImmediate } from "node:timers";

function main(): void {
  const args = process.argv.slice(2);
  const name = args[0] ?? "world";
  const excited = args.includes("--excited");

  let message = `Hello, ${name}`;
  if (excited) {
    message += "!";
  }

  console.log(message);
}

main();
setImmediate(() => {});
